src/librustdoc/clean/mod.rs

```diff
@@ -955,6 +956,42 @@ fn clean_ty_generics<'tcx>(
     }
 }
 
+fn clean_ty_typedef_inner_type<'tcx>(ty: Ty<'tcx>, cx: &mut DocContext<'tcx>) -> TypedefInnerType {
+    if let ty::Adt(adt_def, args) = ty.kind() {
+        if !adt_def.is_struct() {
```

Comment:
Please use [is_enum](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.AdtDef.html#method.is_enum) instead (`is_struct` doesn't return `true` for unions as far as I can see).